### PR TITLE
Respect scale lock when panning the map

### LINKED ISSUE #38953: Scale lock is ignored when panning the map
To reproduce:

1. load world layer from the coordinate line input
2. center to some small island you like
3. set scale to 1:1 000 000
4. lock scale
5. pan the map -> the scale will change

### PR DESCRIPTION
**Pull request by elpaso**
Fixes #38953 and #38417

### CODE REVIEW

**Comment by nyalldawson** [comment]
Would you mind adding a test for this? We don't have much test coverage of the map canvas methods (they've been around forever and easily predate the current test-focused mindset).

**Comment by nyalldawson** [comment]
(fix looks good though!)

**Comment by elpaso** [comment]
@nyalldawson I took a different rout and hopefully fixed a series of bugs with scale locking. Tests added as well.

File: python/gui/auto_generated/qgsmapcanvas.sip.in [hunk 1]
```diff
@@ -1201,6 +1201,7 @@ called when panning is in action, reset indicates end of panning
 called on resize or changed extent to notify canvas items to change their rectangle
 %End
 
+      public:
```
Comment (elpaso) [inline python/gui/auto_generated/qgsmapcanvas.sip.in:1204]:
```suggestion
```

**Review by nyalldawson — APPROVED** [review]
Let's give it a try!

**Pull request merged**